lookout/style/format/feature_extractor.py

```diff
@@ -803,7 +803,8 @@ def _fill_vnode_parents(self, file_parents: Mapping[int, bblfsh.Node],
         for j, vn in enumerate(file_vnodes):
             if vn.node:
                 closest_left_node_id = id(vn.node)
-            parent = self._find_parent(j, file_vnodes, file_parents, closest_left_node_id)
-            if parent is None:
-                parent = uast
-            vnode_parents[id(vn)] = parent
+                vnode_parents[id(vn)] = file_parents[id(vn.node)]
+            else:
+                parent = (self._find_parent(j, file_vnodes, file_parents, closest_left_node_id)
+                          or uast)
+                vnode_parents[id(vn)] = parent
```

Comment:
From the code, it is hard to get how exactly you fix the problem. Can you elaborate a bit?

Reply:
I fix 2 other parent bugs but those do not fix the driver problem itself --- it remains.